Formula/gdal.rb

```diff
@@ -73,35 +73,24 @@ class Gdal < Formula
   fails_with gcc: "5"
 
   def python3
-    "python3.10"
+    "python3.11"
   end
 
   def install
-    args = [
-      "-DENABLE_PAM=ON",
-      "-DCMAKE_INSTALL_RPATH=#{lib}",
-    ]
-    args.concat(std_cmake_args)
-    args_no_python = args.dup << "-DBUILD_PYTHON_BINDINGS=OFF"
+    # Work around Homebrew's "prefix scheme" patch which causes non-pip installs
+    # to incorrectly try to write into HOMEBREW_PREFIX/lib since Python 3.10.
+    inreplace "swig/python/CMakeLists.txt",
+              /(set\(INSTALL_ARGS "--single-version-externally-managed --record=record.txt")\)/,
+              "\\1 --install-lib=#{prefix/Language::Python.site_packages(python3)})"
 
-    mkdir "build" do
-      # First, build without Python to avoid a Linux issue where the
-      # Python bindings are installed in the wrong path
-      # See https://github.com/Homebrew/homebrew-core/pull/116073#issuecomment-1320875424
-      system "cmake", "..", *args_no_python
-      system "make"
-      system "make", "install"
-
-      # Next, reconfigure with Python and manually run the python build
-      args_with_python = args.dup << "-DBUILD_PYTHON_BINDINGS=ON"
-      system "cmake", "..", *args_with_python
-      system "make"
-
-      # Build Python bindings
-      cd "swig/python" do
-        system python3, *Language::Python.setup_install_args(prefix, python3)
-      end
-    end
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DBUILD_PYTHON_BINDINGS=ON",
+                    "-DPython_EXECUTABLE=#{which(python3)}",
+                    "-DENABLE_PAM=ON",
+                    "-DCMAKE_INSTALL_RPATH=#{lib}",
```

Comment:
Absolute path is a little easier for Python bindings.

GDAL has a `GDAL_SET_INSTALL_RELATIVE_RPATH` but it is ignored for Python https://github.com/OSGeo/gdal/blob/v3.6.1/swig/python/CMakeLists.txt#L354-L370.

> The Python bindings target does not currently support setting relative RPATH on the Python wrapper shared libraries. Disable GDAL_SET_INSTALL_RELATIVE_RPATH and set an absolute value for CMAKE_INSTALL_RPATH